aten/src/ATen/QTensorImpl.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include <c10/core/TensorImpl.h>
+#include <ATen/Quantizer.h>
+#include <c10/util/Exception.h>
+
+namespace at {
+
+struct CAFFE2_API QTensorImpl : public c10::TensorImpl {
+public:
+  QTensorImpl(Storage&& storage, TensorTypeId type_id, bool is_variable, std::shared_ptr<at::Quantizer> quantizer);
+
+  at::Quantizer* quantizer() {
```

Comment:
Wouldn't a `const&` be more appropriate here? (Haven't read most of the patch yet, so I don't know; but since your quantizer doesn't have static lifetime I am nervous about returning a raw pointer.)

Reply:
Ideally we should return std::shared_ptr<Quantizer> but another question is how can we expose this to python frontend, I'll probably change this together in that diff.

Reply:
just throw in a //FIXME imo.